src/ui/components/Events/EventsDropDownMenu.tsx

```diff
@@ -110,20 +100,31 @@ function EventTypeContextMenuItem({
   const enabled = filters[category] !== false;
   const checked = enabled && count > 0;
 
+  const onChange = () => {
+    startTransition(() => {
+      setFilters({
+        ...filters,
+        [category]: !enabled,
+      });
+    });
+  };
+
   return (
-    <ContextMenuItem
-      disabled={count === 0 || isPending}
-      onSelect={createSelectHandler(() =>
-        startTransition(() => {
-          setFilters({
-            ...filters,
-            [category]: !enabled,
-          });
-        })
-      )}
-    >
-      <div className={styles.MenuItem}>
-        <Checkbox className={styles.Checkbox} label={category} checked={checked} />
+    <ContextMenuItem disabled={count === 0 || isPending}>
+      <div
+        className={styles.MenuItem}
+        onClick={event => {
+          // Don't close the drop-down menu on click
+          event.preventDefault();
+          event.stopPropagation();
+        }}
+      >
+        <Checkbox
+          checked={checked}
+          className={styles.Checkbox}
+          label={category}
+          onChange={onChange}
+        />
```

Comment:
This was causing a React uncontrolled form control warning. Fixed it as part of this.